docs/clusterpools.md

```diff
@@ -0,0 +1,86 @@
+# Cluster Pools
+
+## Overview
+
+Hive exposes a `ClusterPool` API which allows users to maintain a pool of "hot"
+precreated `ClusterDeployments`, ready to be claimed when needed. The pool size
+can be configured and Hive will attempt to maintain that set number of
+clusters.
+
+When a user needs a cluster they create a `ClusterClaim` resource which will be
+filled immediately with details on where to find their cluster.  (or as soon as
+a cluster is available in the pool if none were presently available). Once
+claimed, a cluster is removed from the pool and a new one will be created to
+replace it. Claimed clusters never return to the pool, they are intended to be
+destroyed when no longer needed. The `ClusterClaim.Spec.Namespace` will be
+populated once the claim has been filled, and the `ClusterDeployment` will be
+present in that namespace with an identical name.
```

Comment:
Yes I am not happy with it when I realized how confusing it looked writing this, but I fear this ship has sailed as it's already published and in use.  It was probably done that way because we try to do that for anything we want to survive a backup and can't easily discover, however in this case we probably could discover it by looking for a cluster matching the label.